addons/roe/roe.lua

```diff
@@ -0,0 +1,280 @@
+-- Copyright © 2017, Cair
+-- All rights reserved.
+
+-- Redistribution and use in source and binary forms, with or without
+-- modification, are permitted provided that the following conditions are met:
+
+    -- * Redistributions of source code must retain the above copyright
+      -- notice, this list of conditions and the following disclaimer.
+    -- * Redistributions in binary form must reproduce the above copyright
+      -- notice, this list of conditions and the following disclaimer in the
+      -- documentation and/or other materials provided with the distribution.
+    -- * Neither the name of ROE nor the
+      -- names of its contributors may be used to endorse or promote products
+      -- derived from this software without specific prior written permission.
+
+-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+-- ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+-- WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+-- DISCLAIMED. IN NO EVENT SHALL Cair BE LIABLE FOR ANY
+-- DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+-- (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+-- LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+-- ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+-- SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+_addon = {}
+_addon.name = 'ROE'
+_addon.version = '1.0'
+_addon.author = "Cair"
+_addon.commands = {'roe'}
+
+packets = require('packets')
+config = require('config')
+require('logger')
+
+
+local defaults = T{
+    profiles = T{
+        default = S{},
+    },
+    blacklist = S{},
+    clear = true,
+    clearprogress = false,
+    clearall = false,
+}
+
+settings = config.load(defaults)
+
+_roe = T{
+    active = T{},
+    complete = T{}    
+}
+
+
+local function cancel_roe(id)
+    id = tonumber(id)
+    
+    if not id then return end
+    
+    if settings.blacklist[id] then return end
+    
+    local p = packets.new('outgoing', 0x10d, {['RoE Quest'] = id })
+    packets.inject(p)
+end
+
+local function accept_roe(id)
+    id = tonumber(id)
+    
+    if not id or _roe.complete[id] or _roe.active[id] then return end
+    
+    local p = packets.new('outgoing', 0x10c, {['RoE Quest'] = id })
+    packets.inject(p)
+end
+
+local function eval(...)
+    assert(loadstring(table.concat({...}, ' ')))()
+end
+
+local function save(name)
+    if not type(name) == "string" then
+        error('`save` : specify a profile name')

```

Comment:
Markdown doesn't actually work here ;D If that was the intent behind using ``.

Since you're using `logger` it will redirect `error` to the FFXI log, so you can use FFXI coloring functions to highlight terms.

Reply:
I think I've avoided color codes forever because they have the rare chance to bug out the chat and I couldn't really pinpoint why